api/src/database/database.py

```diff
@@ -59,7 +75,7 @@ class Database:
 
     def __new__(cls, *args, **kwargs):
         if not isinstance(cls.instance, cls):
-            with lock:
+            with cls.lock:
```

Comment:
[question]: Why the class lock instead of the file level lock?

Reply:
This is a good question, there are two locks in this file, one class level and one file level. We can open a new issue to investigate it.

Reply:
To avoid confusion, we can delete the file level lock if it's not in use.

Reply:
Done.